### PR TITLE
PM rotor

### PR DESCRIPTION
**Pull request by ngadiyar93**

### CODE REVIEW

**Comment by elsevers** [comment]
@ngadiyar93, This branch has conflicting files that must be resolved before merging into develop.

It looks like you still need to merge develop into this branch and clean up.

**Comment by ngadiyar93** [comment]
@elsevers  In addition, I also need to update parts of the code to reflect the new parameterization. I'll complete both and indicate here when I am ready for the PR to be reviewed.

**Comment by ngadiyar93** [comment]
This PR is ready for Review

**Comment by elsevers** [comment]
Drawing does not define a coherent cross-section. Please revise and let us know when you are ready to review.

**Comment by ngadiyar93** [comment]
I have updated the drawing and the code. This PR is ready for review

**Comment by elsevers** [comment]
Please fix issue related to `Location2D`

**Comment by ngadiyar93** [comment]
This PR is now ready for review

**Comment by ngadiyar93** [comment]
I have merged 'develop' into PM Rotor. This fixes the issue I had with location 2D. This PR can be reviewed.

File: model_obj/crosssects/trapezoid/CrossSectTrapezoid.m [hunk 1]
```diff
@@ -32,7 +32,20 @@
             
             y = [0, h, h, 0];
             
+<<<<<<< HEAD
```
Comment (elsevers) [inline model_obj/crosssects/trapezoid/CrossSectTrapezoid.m]: @ngadiyar93 It appears that you have committed a merge conflict into the your branch. Clean this up before merging.

**Review by elsevers — CHANGES_REQUESTED** [review]
Please fix `CrossSectTrapezoid.m`

File: model_obj/crosssects/trapezoid/CrossSectTrapezoid.m (hunk 1 above)
Comment (ngadiyar93) [reply model_obj/crosssects/trapezoid/CrossSectTrapezoid.m]: @elsevers  I have resolved the merge conflict now.

File: model_obj/crosssects/trapezoid/CrossSectTrapezoid.m [hunk 2]
```diff
@@ -34,6 +34,7 @@
             
             [p] = obj.location.transformCoords([x' y']);
 
+
```
Comment (elsevers) [inline model_obj/crosssects/trapezoid/CrossSectTrapezoid.m]: There is a whitespace change to this file. We should see no changes to `CrossSectTrapezoid.m` in this PR. Please fix.

**Review by elsevers — CHANGES_REQUESTED** [review]
Please fix `CrossSectTrapezoid.m` so that there are no changes.

File: model_obj/crosssects/trapezoid/CrossSectTrapezoid.m (hunk 2 above)
Comment (ngadiyar93) [reply model_obj/crosssects/trapezoid/CrossSectTrapezoid.m]: @elsevers  I have now fixed this.

File: model_obj/crosssects/inner_rotor_PMrotor/CrossSectInnerRotorPMRotor.m [hunk 3]
```diff
@@ -0,0 +1,169 @@
+classdef CrossSectInnerRotorPMRotor < CrossSectBase
```
Comment (elsevers) [inline model_obj/crosssects/inner_rotor_PMrotor/CrossSectInnerRotorPMRotor.m]: This structure is no longer actually a PM rotor. Please remove all references to PM rotor. Replace `PMRotor` with `NotchedRotor`

**Review by elsevers — CHANGES_REQUESTED** [review]
Replace all code, folder, and file references to this being a `PMRotor` with `NotchedRotor`

Please make `InnerRotorPMRotor.pdf` more professional looking. The dimensions in particular look distorted. Please use as consistent styling with other cross-sectional drawings as possible.

File: model_obj/crosssects/inner_rotor_PMrotor/CrossSectInnerRotorPMRotor.m (hunk 3 above)
Comment (ngadiyar93) [reply model_obj/crosssects/inner_rotor_PMrotor/CrossSectInnerRotorPMRotor.m]: @elsevers I have removed all references to a PM rotor and replaced 'PMRotor' with 'NotchedRotor'. I have also enhanced the figure.

**Review by npetersen2 — DISMISSED** [review]
Pending Eric's approval, this seems good to me.

**Comment by ngadiyar93** [comment]
I have updated the dimension 'alpha' in the figure to be clear, using inkscape's native fonts. This PR is ready for review.

**Review by elsevers — APPROVED** [review]
Looks good

**Review by npetersen2 — APPROVED** [review]
Looks good to me!

**Pull request merged**